flask_security/core.py

```diff
@@ -245,11 +247,24 @@ def _request_loader(request):
         if isinstance(data, dict):
             token = data.get(args_key, token)
 
+    use_cache = current_app.config.get('USE_VERIFY_PASSWORD_CACHE')
+
     try:
         data = _security.remember_token_serializer.loads(
             token, max_age=_security.token_max_age)
         user = _security.datastore.find_user(id=data[0])
-        if user and verify_hash(data[1], user.password):
+
+        has_cache = False
+        if use_cache:
+            cache = getattr(local_cache, 'verify_hash_cache', None)
+            if cache is None:
+                cache = VerifyHashCache()
+                local_cache.verify_hash_cache = cache
+            has_cache = cache.has_verify_hash_cache(token, user)
```

Comment:
Nit: this is fairly convoluted logic just to avoid replicating a line or 2 of code. And you aren't handing the 'not user' case.
```suggestion
            if not user:
                return _security.login_manager.anonymous_user()
            if use_cache:
                  cache = getattr(local_cache, 'verify_hash_cache', None)
                  if cache is None:
                       cache = VerifyHashCache()
                       local_cache.verify_hash_cache = cache
                  if cache.has_verify_hash_cache(token, user):
                      return user
                  if verify_hashdata[1], user.password):
                      cache.set_cache(token, user)
                      return user
              else:
                   if verify_hashdata[1], user.password):
                       return user
                      
                      
```

Reply:
I implented this at first glance, but did not change it until you notice it. It is now changed as you suggested and it is indeed clearer the way you suggested. Thank you for this too !